nestkernel/node_collection.h

```diff
@@ -218,6 +218,19 @@ class NodeCollection
    */
   static NodeCollectionPTR create( const std::vector< index >& node_ids );
 
+  /**
+   * @brief Create a NodeCollection from an array of node IDs without requiring
+   * instantiation of a node model and associated managers. Intended to simplify
+   * C++ test programs where just a NodeCollection is needed. Since the fingerprint
+   * of the NodeCollection superclass require a kernel instance, a kernel must be
+   * instantiated, i.e., KernelManager::create_kernel_manager(), before this
+   * creator can be used.
```

Comment:
Maybe add a @note that these mock collections can only be use to test certain properties of node collections.